### PR TITLE
Fix potential passing of NoneType to remove_ansi_escape

### PR DESCRIPTION
**Pull request by u007**
seems to be a logging issue

### CODE REVIEW

**Review by lc0rp — COMMENTED** [review]
Nice! Could you pair it w/ a unit test?

File: autogpt/logs.py [hunk 1]
```diff
@@ -254,6 +254,8 @@ def remove_color_codes(s: str) -> str:
 
 
 def remove_ansi_escape(s: str) -> str:
+    if s is None:
+        return ""
```
Comment (Pwuts) [inline autogpt/logs.py]: This is invalid considering the type annotation of `str`. A cleaner solution would be to make sure a `str` is always passed, see below.

File: autogpt/logs.py [hunk 2]
```diff
@@ -270,8 +272,12 @@ def print_assistant_thoughts(
     assistant_thoughts_speak = None
     assistant_thoughts_criticism = None
 
-    assistant_thoughts = assistant_reply_json_valid.get("thoughts", {})
-    assistant_thoughts_text = remove_ansi_escape(assistant_thoughts.get("text"))
+    assistant_thoughts = assistant_reply_json_valid.get("thoughts", None)
+    assistant_thoughts_text = (
+        remove_ansi_escape(assistant_thoughts.get("text"))
```
Comment (Pwuts) [inline autogpt/logs.py]:
```suggestion
        remove_ansi_escape(assistant_thoughts.get("text", ""))
```

**Review by Pwuts — CHANGES_REQUESTED** [review]

**Review by Pwuts — APPROVED** [review]

**Comment by Pwuts** [comment]
Note that this PR doesn't fix the instance reported in #4845:

      File "C:\Users\Zachary\Documents\Personal_Projects\autogpt\auto-gpt_git\Auto-GPT\autogpt\agent\agent.py", line 188, in start_interaction_loop
        f"COMMAND = {Fore.CYAN}{remove_ansi_escape(command_name)}{Style.RESET_ALL}  "
      File "C:\Users\Zachary\Documents\Personal_Projects\autogpt\auto-gpt_git\Auto-GPT\autogpt\logs.py", line 253, in remove_ansi_escape
        return s.replace("\x1B", "") AttributeError: 'NoneType' object has no attribute 'replace'

**Pull request merged**